pkg/geoip/geoip.go

```diff
@@ -18,6 +20,22 @@ type Location struct {
 
 // Fetch retrieves the location of the system calling this function
 func Fetch() (Location, error) {
+	geoipURLs := []string{"https://geoip.grid.tf/", "https://02.geoip.grid.tf/", "https://03.geoip.grid.tf/"}
+
+	for _, url := range geoipURLs {
+		l, err := getLocation(url)
+		if err != nil {
+			log.Err(err).Msgf("failed to fetch location from geoip service %s. retrying...", url)
```

Comment:
the log here is kinda misleading, we are not `retrying` instead we are trying a completely different url. Instead the error should show which url is not working + the cause. Also please use the structured logs correctly. Use the available, Str, Int, etc.. not formatting as much as possiblel

```go
log.Err(err).Str("url", url).Msg("failed to fetch location from geoip service")
```

Is more correct form than the one in code